MatrixSDK/MXSession.m

```diff
@@ -1343,6 +1390,12 @@ - (void)serverSyncWithServerTimeout:(NSUInteger)serverTimeout
                 return;
             }
             
+            //  clear initial sync cache after handling sync response
+            for (NSString *responseId in self.initialSyncResponseCache.outdatedSyncResponseIds)
+            {
+                [self.initialSyncResponseCache deleteSyncResponseWithId:responseId];
```

Comment:
It is too early. We need to wait for the completion of async operations launched by MXFileStore.commit().
We are missing an interface in MXStore to be able to synchronise on the end of those asynchronous operations.

One way to test it is to create a crash or restart your app just after this line. On a real account, MXFileStore will be empty on the next startup.

Reply:
Thanks, updated.